src/search.coffee

```diff
@@ -6,234 +6,252 @@ url = require 'url'
-        id: entry.ID
-        title: entry.Title
-        url: entry.MediaUrl
-        displayUrl: entry.DisplayUrl
-        runtime: Number entry.RunTime
-        thumbnail: @extractThumbnail entry
-
-  news: (query, options, callback) ->
-    @verticalSearch 'News', _.bind(@extractNewsResults, this), query, options,
-      callback
-
-  extractNewsResults: (results) ->
-    @mapResults results, (entry) ->
-      id: entry.ID
-      title: entry.Title
-      source: entry.Source
-      url: entry.Url
-      description: entry.Description
-      date: new Date entry.Date
-
-  spelling: (query, options, callback) ->
-    @verticalSearch 'SpellingSuggestions', _.bind(@extractSpellResults, this),
-      query, options, callback
-
-  extractSpellResults: (results) ->
-    @mapResults results, ({Value}) ->
-      Value
-
-  related: (query, options, callback) ->
-    @verticalSearch 'RelatedSearch', _.bind(@extractRelatedResults, this),
-      query, options, callback
-
-  extractRelatedResults: (results) ->
-    @mapResults results, ({Title, BingUrl}) ->
-      query: Title
-      url: BingUrl
-
-  composite: (query, options, callback) ->
-    [callback, options] = [options, {}] if arguments.length is 2
-    options = _.defaults {}, options, {query, sources: Search.SOURCES}
-
-    @parallelSearch 'Composite', options, (err, results) =>
-      return callback err if err
 
-      convertToSingleSource = (results, source) ->
-        {d: {results: r.d.results[0][source]}} for r in results
+        data =
+          estimatedCount: _.last(responses).estimatedCount
+          results: []
+
+        # Avoid duplicates by checking result IDs.
+        existingIds = []
+        responses.forEach (response) ->
+          response.results.forEach (result) ->
+            unless result.id in existingIds
+              data.results.push result
+              existingIds.push result.id
+
+        callback null, data
+
+  counts: (query, options..., callback) ->
+    sources = []
+
+    sources.push
+      key: 'web'
+      method: _.bind @rawWeb, this
+    sources.push
+      key: 'image'
+      method: _.bind @rawImages, this
+    sources.push
+      key: 'video'
+      method: _.bind @rawVideos, this
+    sources.push
+      key: 'news'
+      method: _.bind @rawNews, this
+
+    executeSearchForCounts = (source, callback) ->
+      ###
+      We only need to one result (zero is impossible) from most verticals. With
+      a web search, however, the `totalEstimatedMatches` needs to be checked
+      from a higher page for accurate data.
+      
+      The 1,000 value comes from empirical data. It seems that after 600
+      results, the accuracy gets quite consistent and accurate. I picked 1,000
+      just to be in the clear. It also doesn't matter if there are fewer than
+      1,000 results.
+      ###
+      pagination = {offset: 0, top: 1}
+      if source.key is 'web'
+        _.each pagination, (value, key) ->
+          pagination[key] += 1000
```

Comment:
Only time it's ever used. Sadly, can't take credit for the explanation -- Dudley wrote that. I did confirm that `totalEstimatedMatches` accuracy _still_ grows for web searches when `offset > 1000`.